packages/web-app-admin-settings/src/components/Users/SideBar/EditPanel.vue

```diff
@@ -170,6 +182,46 @@ export default defineComponent({
     changeSelectedGroupOption(option: Group) {
       this.editUser.memberOf = option
     },
+    async validateUserName() {
+      this.formData.userName.valid = false
+
+      if (this.editUser.onPremisesSamAccountName.trim() === '') {
+        this.formData.userName.errorMessage = this.$gettext('User name cannot be empty')
+        return false
+      }
+
+      if (this.editUser.onPremisesSamAccountName.includes(' ')) {
+        this.formData.userName.errorMessage = this.$gettext('User name cannot contain white spaces')
+        return false
+      }
+
+      if (
+        this.editUser.onPremisesSamAccountName.length &&
+        !isNaN(this.editUser.onPremisesSamAccountName[0])
+      ) {
+        this.formData.userName.errorMessage = this.$gettext('User name cannot start with a number')
+        return false
+      }
+
+      if (this.user.onPremisesSamAccountName !== this.editUser.onPremisesSamAccountName) {
+        try {
+          await this.graphClient.users.getUser(this.editUser.onPremisesSamAccountName)
+          this.formData.userName.errorMessage = this.$gettext(
+            this.$gettext('User "%{userName}" already exists'),
+            { userName: this.editUser.onPremisesSamAccountName }
+          )
```

Comment:
Now you have 2 `$gettext` 😄 

```suggestion
          this.formData.userName.errorMessage = this.$gettext(
            'User "%{userName}" already exists',
            { userName: this.editUser.onPremisesSamAccountName }
          )
```

Reply:
i see, there is more to do